internal/replicationslot/repo/repo.go

```diff
@@ -0,0 +1,113 @@
+package repo
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+	"strings"
+
+	"reactive-tech.io/kubegres/internal/replicationslot"
+)
+
+var (
+	ErrAlreadyExist = errors.New("slot already exist")
+	ErrDoesNotExist = errors.New("does not exist")
+)
+
+type Repository interface {
+	CreateSlot(ctx context.Context, name string) (replicationslot.ReplicationSlot, error)
+	FindSlotByName(ctx context.Context, name string) (replicationslot.ReplicationSlot, error)
```

Comment:
just a nit:
```suggestion
	GetSlot(ctx context.Context, name string) (replicationslot.ReplicationSlot, error)
```
I think `name` is the ID of the slot, so we can make it easier to understand this API.
If we add some fins slots by other filters, then we can add `Find*` methods.